##### Nafisa Tasnia
#### DS3000
#### 1/28/2024

In [2]:
import numpy as np

In [3]:
# Q1:

In [4]:
# Loading data
data = np.loadtxt("birthwt.csv", delimiter=',', skiprows=1)

In [5]:
# Turns data into an array
data_array = np.array(data, dtype=float)

In [256]:
# Displays the dimension of the data
print("Dimension is:", data_array.shape)

Dimension is: (189, 10)


In [257]:
# Q2:
# The age range of mothers in this dataset
age = data_array[:,1]

# Finds the min,max, and average ages
age_min = np.min(age)
age_max = np.max(age)
age_average = (np.round(np.mean(age)))

# displays the min, max, and average ages
print("The min age is:", age_min)
print("The max age is:", age_max)
print("The average age is:", age_average)



The min age is: 14.0
The max age is: 45.0
The average age is: 23.0


In [258]:
# Question 3: How many mothers smoked in pregnancy
# specifically gets coloumn 4 where the data about whether the mother smoked during pregnancy or not
smoking = data_array[:,4]

In [259]:
# counts how many mothers smoked during pregnancy 
do_smoke = smoking == 1
do_count = np.sum(do_smoke)

In [260]:
# displays the number of mothers that smoked
print("The number of mothers that smoked:", do_count)

The number of mothers that smoked: 74


In [261]:
# Q4: How many mothers had a history of hypertension?
# gets the coloumn that contains info about hypertension history of the mothers
ht = data_array[:,6]

In [262]:
# finds all the mothers who has hypertension and counts them
has_hypers = ht == 1
hyper_counter = np.sum(has_hypers)

In [263]:
# displays how many mothers have hypertension 
print("number of mothers with hypertension:", hyper_counter)

number of mothers with hypertension: 12


In [279]:
#Q5 How many mothers have 1 or more premature births?
                        
# gets the coloumn that contains data about premature births
plt = data_array[:, 5]

# counts how many pre-mature births there are 
is_premature = plt > 0
total_premature = np.sum(is_premature)
                        
# displays number of mothers who have 1 or more premature births?                       
print("The number of mothers who have 1 or more premature births", total_premature)


The number of mothers who have 1 or more premature births 30


In [280]:
# Q6 Create a function called convert_birthweight

# Establishes variable that has the numbers needed during the conversion of grams to pounds to ounces
one_pound = 453.6
one_ounce = 16

def convert_birthweight(grams):
    '''
    function: converts grams to pounds and the remaining into ounces rounded
    parameter: 
    (grams): takes in grams, the measure of unit
    '''
    pounds = grams / one_pound 
    final_pounds = int(pounds)
    remaining_ounces = (pounds - final_pounds) * one_ounce  
    return f"{final_pounds} pounds and {remaining_ounces:.2f} ounces"


In [278]:
#Q7 What is the average birth weight of babies whose mothers have a history of premature birth? 

# finds only the weight of babies whoses mothers hvae a history of premature birth
# converting the weight into pounds and ounces 
if total_premature > 0:
    birth = data_array[is_premature, 9]
    birth_total = np.mean(birth)
    converting_birth = (convert_birthweight(birth_total))
    
    # prints results
    print("The average birth weight of babies whose mothers have a history of premature birth is:", 
          converting_birth)


The average birth weight of babies whose mothers have a history of premature birth is: 5 pounds and 10.98 ounces


In [267]:
# Q8 What is the minimum, average and maximum birth weight of babies
# whose mother did not smoke during pregnancy and do not have a history of hypertension?

In [268]:
# finds number of mothers that don't have hypertension
hyper = ht == 0
hyper_counter = np.sum(hyper)

# finds number of monthers who don't smoke 
non_smoker = smoking == 0
no = np.sum(non_smoker)

# merges the data to create a new array
mix = data_array[hyper & non_smoker]

# Extract birth weight column for mothers without hypertension and non-smokers
births = (mix[:, 9])

# finds the min, max, and mean of weight of babies whose mothers don't smoke and have no hypertension
find_mn = (convert_birthweight(np.min(births)))
find_mx = (convert_birthweight(np.max(births)))
find_average = (convert_birthweight(np.mean(births)))

print("The minimum weight of babies whose mothers don't smoke and have no hypertension is:",find_mn + ".")
print("The max weight of babies whose mothers don't smoke and have no hypertension is:", find_mx + ".")
print("The average weight of babies whose mothers don't smoke and have no hypertension is:", find_average + ".")




The minimum weight of babies whose mothers don't smoke and have no hypertension is: 2 pounds and 4.01 ounces.
The max weight of babies whose mothers don't smoke and have no hypertension is: 11 pounds and 0.01 ounces.
The average weight of babies whose mothers don't smoke and have no hypertension is: 6 pounds and 13.01 ounces.


In [269]:
# Q9 What is the minimum, average and maximum birth weight of babies whose mothers smoked during pregnancy?

# finds all mothers who smoked during pregnancy
do_smoke = smoking == 1
do_count = np.sum(do_smoke)

# finds the associated weights of babies whose mothers smoked during pregnancy 
new_d = data_array[do_smoke, 9]

# finds the min, max, and average weights of babies whose mothers smoked during pregnancy    
finding_mn = (convert_birthweight(np.min(new_d)))
finding_mx = (convert_birthweight(np.max(new_d)))
finding_average = (convert_birthweight(np.mean(new_d)))

# displays the min, max, and average weight of the babies
print("The minimum birthweight of babies whose mothers smoked during pregnancy is", finding_mn + ".")
print("The maximum birthweight of babies whose mothers smoked during pregnancy is", finding_mx + ".")
print("The average birthweight of babies whose mothers smoked during pregnancy is", finding_average + ".")



The minimum birthweight of babies whose mothers smoked during pregnancy is 1 pounds and 9.01 ounces.
The maximum birthweight of babies whose mothers smoked during pregnancy is 9 pounds and 5.49 ounces.
The average birthweight of babies whose mothers smoked during pregnancy is 6 pounds and 1.77 ounces.


In [274]:
# Q10 What is the minimum, average and maximum birth weight of babies whose mothers had 
# a history of hypertension?

# finds the associated babies wieght of the mothers who have hyptensions 
new_h = data_array[has_hypers, 9]

# finds min, max, and average birth weight of babies whose mom had a history of hypertension
finding_mns = (convert_birthweight(np.min(new_h)))
finding_mxs = (convert_birthweight(np.max(new_h)))
find_average = (convert_birthweight(np.mean(new_h)))

# displays the min, max, and average birth weights of babies whose mom had a history of hypertension 
print("The minimum birth weight of babies whose mothers had a history of hypertension is", finding_mns + ".")
print("The maximum birth weight of babies whose mothers had a history of hypertension is", finding_mxs + ".")
print("The average birth weight of babies whose mothers had a history of hypertension is", find_average + ".")


The minimum birth weight of babies whose mothers had a history of hypertension is 2 pounds and 8.04 ounces.
The maximum birth weight of babies whose mothers had a history of hypertension is 8 pounds and 5.69 ounces.
The average birth weight of babies whose mothers had a history of hypertension is 5 pounds and 9.48 ounces.


In [276]:
# Q11: What is the minimum, average and maximum birth weight of babies whose
# mothers smoked during pregnancy and had a history of hypertension? 

# creates new array to only contain mothers who smoke and have hypertension 
new_h_s = data_array[do_smoke & has_hypers]

# finds those mothers baby weight
birthing = new_h_s[:, 9]

# finds min, max, and average weight of babies whose mothers smoked and had a history of hypertension
find_min = (convert_birthweight(np.min(birthing)))
find_max = (convert_birthweight(np.max(birthing)))
find_average = (convert_birthweight(np.mean(birthing)))

# displays the min,max, and average 
print("The minimum weight of babies whose mothers smoked during pregnancy and had a history of hypertension is:", 
      find_min)
print("The maximum weight of babies whose mothers smoked during pregnancy and had a history of hypertension is:", 
      find_max)
print("The average weight of babies whose mothers smoked during pregnancy and had a history of hypertension is:", 
      find_average)

The minimum weight of babies whose mothers smoked during pregnancy and had a history of hypertension is: 2 pounds and 8.04 ounces
The maximum weight of babies whose mothers smoked during pregnancy and had a history of hypertension is: 8 pounds and 4.49 ounces
The average weight of babies whose mothers smoked during pregnancy and had a history of hypertension is: 5 pounds and 10.34 ounces


#### Q12: ) Compare and contrast the results from 7-11 above and summarize your findings. Do you see any patterns in the birth weight of babies whose mother did not have any history of hypertension and/or did not smoke during pregnancy versus those who did? Ensure that you reference specific birth weights to support your explanation.


After analyzing all the data, one pattern that can be observed is that babies born to mothers who didn't smoke during pregnancy and have no history of hypertension tend to have a higher average birth weight of 6 pounds and 13.01 ounces compared to mothers who have a history of hypertension and smoking during pregnancy. As the average birth weight for them is 5 pounds and 10.34 ounces. Mothers who didn't have a history of hypertension and didn't smoke during pregnancy had the highest average. Now comparing the maximum weight we can see mothers who don't smoke and don't have hypertension can have babies who weigh up to 11 pounds and 0.01 ounces while the maximum weight for babies whose mothers did smoke during their pregnancy and do have a history of hypertension is 8 pounds and 4.49 ounces. So smoking and having a history of hypertension can result in lower birth weight for the baby. From this, it can be inferred that a mother is more likely to produce a healthier baby in terms of weight if they don't smoke during pregnancy and don't have a history of hypertension.

One commonality that can be concluded from this analysis is the presence of low birth weights ranging from 1 pound and 9.01 ounces to 2 pounds and 8.04 ounces. This means that low birth weight is observed across all groups. Then looking at the maximum birth weights, it varies from each group. For example, the max birth weight of babies for mothers with a history of hypertension was 8 pounds and 5.69 ounces, while the max weight of babies of mothers who didn't smoke during pregnancy and didn't have hypertension was 11 pounds and 0.01 ounces. This once again shows that a baby has the potential to be at a healthier weight if the mother doesn't have a history of hypertension and doesn't smoke during pregnancy.

In conclusion, maternal lifestyle factors like hypertension history and smoking during pregnancy play a role in the weight of babies. Smoking and having a history of hypertension can increase the chance of a baby weighing less compared to mothers who abstain from smoking during pregnancy and mothers who don't have any history of hypertension.
